File: Chavah/Models/Album.cs
using System;

namespace BitShuva.Models
{
    /// <summary>
    /// An album for which album art has been uploaded.
    /// </summary>
    public class Album
    {
        public string Artist { get; set; }
        public string Name { get; set; }
        public Uri AlbumArtUri { get; set; }
        public string Id { get; set; }
        public string BackgroundColor { get; set; }
        public string ForegroundColor { get; set; }
        public string MutedColor { get; set; }
        public string TextShadowColor { get; set; }
        public bool IsVariousArtists { get; set; }
        public int SongCount { get; set; }

        public System.Linq.Expressions.Expression<Func<Song, bool>> SongMatchesAlbumNameAndArtistCriteria()
        {
            if (this.IsVariousArtists)
            {
                return s => s.Album == this.Name;
            }
            else
            {
                return s => s.Album == this.Name && s.Artist == this.Artist;
            }
        }
    }
}

File: Chavah/App/Models/Album.ts
namespace BitShuva.Chavah {
    export class Album implements Server.IAlbum {
        artist: string;
        name: string;
        albumArtUri: string | null;
        id: string;
        backgroundColor: string;
        foregroundColor: string;
        mutedColor: string;
        textShadowColor: string;
        isVariousArtists: boolean;
        songCount: number;

        isSaving = false;

        constructor(serverObj: Server.IAlbum) {
            angular.merge(this, serverObj);
        }
    }
}